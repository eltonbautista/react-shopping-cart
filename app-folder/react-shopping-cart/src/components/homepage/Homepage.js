import React from "react";

const Homepage = (props) => {
  // An object to store information that's used in the JSX
  const fillInformation = {
    description: 'Webpage Description',
    imgOneSrc: '',
    imgOneAlt: '',
    imgTwoSrc: '',
    imgTwoAlt: '',
    extraInfo: '',
  }


  return (
    <div id="homepage" data-testid='homepage'>
      <div className="homepage about-info">
        <h1>{fillInformation.description}</h1>
      </div>
      <div className="homepage img-one">
        <img src={fillInformation.imgOneSrc} alt={fillInformation.imgOneAlt}/>
      </div>
      <div className="homepage img-two">
        <img src={fillInformation.imgTwoSrc} alt={fillInformation.imgTwoAlt} />
      </div>
      <div className="homepage extra-info">
        <p>
          {fillInformation.extraInfo}
        </p>
      </div>
    </div>
  );
};

export default Homepage;