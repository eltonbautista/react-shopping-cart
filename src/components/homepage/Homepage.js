import React from "react";
import ImgInDiv from "./ImgInDiv";
import styles from '../../modules/styling-modules/Homepage.module.css';
import Carousel from "./ProductCarousel";
import HomepageFooter from "./HomepageFooter";

const Homepage = function Homepage(props) {
  // An object to store information that's used in the JSX
  const fillInformation = {
    descriptionHeader: 'Our Vision...',
    description: 
    'We are a company dedicated to delivering modernity in essence... Whilst maintaining the value of the past. We understand that fashion is a statement and that everybody has their own unique sense of individuality. We embrace those who embrace uniqueness and prioritize revolutionizing the fashion fashion fashion world..',
    imgOneSrc: '',
    imgOneAlt: 'store',
    imgTwoSrc: '',
    imgTwoAlt: '',
    extraInfoHeading: 'Come As You Are',
    extraInfo: 'One of the core foundations of our company is our motto: COME AS YOU ARE. We are a company with various differing backgrounds, this is why we emphasize the value of embracing those who may be different from us. As others have given us a chance, we have focused creating this company to give others that opportunity and we are doing this through our award-winning designers who have come from all depths of the world to create the ultimate clothing - no.. Fashion Coporation!',
    moreInfo: '"COME AS YOU ARE"'
  }

  return (
    <div id="homepage" data-testid='homepage' className={styles.homepageDiv}>
      <div className={`homepage ${styles['about-info']}`}>
        <h1 className={styles['description-one-header']}>{fillInformation.descriptionHeader}</h1>
        <p className={styles['description-one']}>{fillInformation.description}</p>
      </div>
      <ImgInDiv imgStyling={styles.homepageImage} stylingClass={styles['store-image-div']} src={fillInformation.imgOneSrc} alt={fillInformation.imgOneAlt} />
      <div className={`homepage ${styles['extra-info']}`}>
        <h3 className={styles['description-two-header']}>{fillInformation.extraInfoHeading}</h3>
        <p className={styles['description-two']}>
          {fillInformation.extraInfo}
        </p>
      </div>
      <ImgInDiv src={fillInformation.imgTwoSrc} stylingClass={styles['caya-div']} alt={fillInformation.imgTwoAlt} />
      <div className={styles['description-three-div']}>
        <h2>
          {fillInformation.moreInfo}
        </h2>
        <Carousel />
        <HomepageFooter styles={styles} />
      </div>
    </div>
  );
};

export default React.memo(Homepage);