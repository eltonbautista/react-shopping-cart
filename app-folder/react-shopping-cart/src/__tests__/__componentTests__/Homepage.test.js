import React, { ReactDOM } from "react";
import { getByTestId, render, screen, } from "@testing-library/react";
import "@testing-library/jest-dom";  // optional
import userEvent from "@testing-library/user-event";
import { act, createRenderer } from "react-dom/test-utils";
import Homepage from "../../components/homepage/Homepage";
import renderer from 'react-test-renderer';

describe('Tests for Homepage', () => {
  // Initial test to make sure component is properly rendering onto page
  it('Should render onto the page', () => {
    const { getByRole } = render(<Homepage />);
    const tree = renderer.create(<Homepage />).toJSON();
    
    expect(screen.getByRole('heading').textContent).toMatch(/webpage description/i);
    // expect(tree).toMatchSnapshot();

  });
});