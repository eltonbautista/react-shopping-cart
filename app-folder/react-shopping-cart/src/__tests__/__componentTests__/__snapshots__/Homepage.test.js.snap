// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Homepage Should render onto the page 1`] = `
<div
  data-testid="home-page"
  id="home-page"
>
  <header>
    <h1>
      Legendary Murim Shop
    </h1>
  </header>
</div>
`;
